bootstrap/kubeadm/api/v1alpha3/kubeadmbootstrapconfig_types.go

```diff
@@ -73,6 +73,11 @@ type KubeadmConfigSpec struct {
 	// It overrides the `--v` flag in kubeadm commands.
 	// +optional
 	Verbosity *int32 `json:"verbosity,omitempty"`
+
+	// UseExperimentalRetryJoin replaces a basic kubeadm command with a shell
+	// script with retries for control plane joins
+	// +optional
+	UseExperimentalRetryJoin bool `json:"useExperimentalRetryJoin,omitempty"`
```

Comment:
```suggestion
	// UseExperimentalRetryJoin replaces a basic kubeadm command with a shell
	// script with retries for control plane joins.
	// 
	// This is meant to be an experimental temporary workaround on some environments
	// where joins fail due to timing (and other issues). The long term goal is to add retries to 
	// kubeadm proper and use that functionality. 
	// For more information, refer to https://github.com/kubernetes-sigs/cluster-api/pull/2763#discussion_r397306055.
	// +optional
	UseExperimentalRetryJoin bool `json:"useExperimentalRetryJoin,omitempty"`
```

Reply:
Also might be worthwhile to add how much this impacts on the overall size of the user data?